src/reduceRight.js

```diff
@@ -20,24 +20,42 @@ var _curry3 = require('./internal/_curry3');
  * @memberOf R
  * @since v0.1.0
  * @category List
- * @sig (a,b -> a) -> a -> [b] -> a
- * @param {Function} fn The iterator function. Receives two values, the accumulator and the
- *        current element from the array.
+ * @sig (a, b -> b) -> b -> [a] -> b
+ * @param {Function} fn The iterator function. Receives two values, the current element from the array
+ *        and the accumulator.
  * @param {*} acc The accumulator value.
  * @param {Array} list The list to iterate over.
  * @return {*} The final, accumulated value.
- * @see R.addIndex
+ * @see R.reduce, R.addIndex
  * @example
  *
- *      var pairs = [ ['a', 1], ['b', 2], ['c', 3] ];
- *      var flattenPairs = (acc, pair) => acc.concat(pair);
+ *      R.reduceRight(R.subtract, 0, [1, 2, 3, 4]) // => (1 - (2 - (3 - (4 - 0)))) = -2
+ *              2
+ *               \
+ *            1 - 3
+ *                 \
+ *              2 - -1
+ *                   \
+ *                3 - 4
+ *                     \
+ *                  4 - 0
+ *
+ *      R.reduce(R.subtract, 0, [1, 2, 3, 4]) // => ((((0 - 1) - 2) - 3) - 4) = -10
+ *                  -10
+ *                 /
+ *                -6 - 4
+ *               /
+ *              -3 - 3
+ *             /
+ *            -1 - 2
+ *           /
+ *          0 - 1
```

Comment:
I find these diagrams a bit difficult to follow.  How about something like

```
  -              -2        
 / \             / \        
1   -           1   3       
   / \             / \      
  2   -    ==>    2  -1    
     / \             / \    
    3   -           3   4   
       / \             / \  
      4   0           4   0 
```
and
```
        -              -10
       / \             / \
      -   4          -4   4
     / \             / \
    -   3    ==>   -3   3
   / \             / \
  -   2          -1   2
 / \             / \
0   1           0   1
```
?  Also, I think the `reduce` example belongs with the `reduce` documentation.  Those who want to compare them can switch back and forth, especially since the two functions link to one another.